core/src/main/java/org/apache/calcite/sql/SqlFunction.java

```diff
@@ -291,6 +291,19 @@ private RelDataType deriveType(
             }
           }
         }
+
+        if (function == null) {
```

Comment:
The outer layer already has the judgment of `if (function == null)` Do we still need to judge here?

Reply:
I dont think so. Done :)

Reply:
> I dont think so. Done :)

Do you mean this [line](https://github.com/apache/calcite/blob/532126a152622aac0f3ccc193dbdda2042590958/core/src/main/java/org/apache/calcite/sql/SqlFunction.java#L281) will also be null?

Reply:
It seems like it could be null